Added Press Back Button Again to Exit App Functionality

Prakhar-Agarwal-byte opened the pull request:
@cpg 
I have added the feature as requested in the issue. Now, first, a toast pops up asking the user to press the back button again, and if the user presses the back button again in 2 seconds, only then the app exit. Fixes #730.

Commit:
Merge pull request #1 from amahi/beta

Update my repo with amahi latest

Commit:
Added confirmation for exiting app.

chirag-jn commented on src/main/res/values/strings.xml:
```diff
@@ -344,5 +344,6 @@
     <string name="size_largest">Size (largest first)</string>
     <string name="size_smallest">Size (smallest first)</string>
     <string name="file_type">File type</string>
+    <string name="message_toast_press_back_again">Press back again to exit</string>
```
Please add translations in other languages.

prakharagarwal-dev replied:
Done

Prakhar-Agarwal-byte replied:
Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine.

prakharagarwal-dev replied:
Please approve changes

prakharagarwal-dev replied:
@chirag-jn 
I have done the required changes already. Please approve the PR.

chirag-jn committed:
Update backPressedTime initialization

If the user presses the back button within 2 seconds of opening the app, they shouldn't receive a prompt for pressing back again.

chirag-jn commented:
Made some changes in the code @Prakhar-Agarwal-byte. Please pull and complete the requested changes.

Commit:
Added translations for back message

Commit:
Added translations for back message

Prakhar-Agarwal-byte commented:
@chirag-jn
Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?

chirag-jn commented:
> @chirag-jn
> Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?

Welcome to Amahi server works fine for me. Try reinstalling the app or creating new credentials on the Amahi website? @Prakhar-Agarwal-byte

Prakhar-Agarwal-byte commented:
> > @chirag-jn
> > Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?
> 
> Welcome to Amahi server works fine for me. Try reinstalling the app or creating new credentials on the Amahi website? @Prakhar-Agarwal-byte

I guess there was some problem with authentication. I signed out and logged in again on the website and it started worked again. Thanks!

Prakhar-Agarwal-byte commented:
> > @chirag-jn
> > Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?
> 
> Welcome to Amahi server works fine for me. Try reinstalling the app or creating new credentials on the Amahi website? @Prakhar-Agarwal-byte

I guess there was some problem with authentication. I signed out and logged in again on the website and it started worked again. Thanks!

cpg commented:
We have had some issues with certificate renewals that caused connection errors. It all should be working well now.
